tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php

```diff
@@ -602,4 +602,96 @@ public function doFoo(iterable $bar, ?int $baz): ?array {}
             ],
         ];
     }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixWithAllowedMixedCases
+     */
+    public function testFixWithAllowedMixed($expected, $input = null)
+    {
+        $this->fixer->configure(['allow_mixed' => true]);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithAllowedMixedCases()
+    {
+        $cases = $this->provideFixCases();
+        $cases['no typehint mixed'] = [
+            '<?php
+class Foo {
+    /**
+     * @param mixed $bar
+     *
+     * @return mixed
+     */
+    public function doFoo($bar) {}
+}',
+        ];
+
+        return $cases;
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixPhp70WithAllowedMixedCases
+     * @requires PHP 7.0
+     */
+    public function testFixPhp70WithAllowedMixed($expected, $input = null)
+    {
+        $this->fixer->configure(['allow_mixed' => true]);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixPhp70WithAllowedMixedCases()
+    {
+        $cases = $this->provideFixPhp70Cases();
+        $cases['no typehint mixed'] = [
```

Comment:
This test is the same as the one added above which has no PHP 7.0 specific syntax. You can drop this one. Same for the PHP 7.1 test below.

Reply:
Wel.. it uses the original provider and just upsert a `no typehint mixed` case, so all previous cases must pass as well - meaning, I didn't break anything by this option..

Should I still drop these tests?

Reply:
I think that for PHP 7.0 and 7.1 tests using `allow_mixed => true`, the extra `no typehint mixed` case isn't necessary (already tested in `testFixWithAllowedMixed()`) so you can drop the specific data providers and use the existing ones directly.

Reply:
Oh you meant _drop providers_ not drop tests.. I makes sense now :+1: